### PR TITLE
Improve documentation searchability

### PR DESCRIPTION
**Pull request by smockle**
This PR:
- Adds alternate search terms to frontmatter
- Links to Octicons docs

Example:
The CounterLabel component is now included in search results for Pill

#### Merge checklist
- [x] Changed base branch to release branch
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [x] Tested in Safari
- [ ] Tested in Edge

### CODE REVIEW

File: docs/content/CounterLabel.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 ---
 title: CounterLabel
+tags:
```
Comment (emplums) [inline docs/content/CounterLabel.md]: Would you be down to format it with all the tags on the same line like: 
```
tags: pill, other tag, etc
```

Comment (smockle) [reply docs/content/CounterLabel.md]: Sure, I’ll submit an update! 👍

It looks like Doctocat is using Gatsby under the hood, and Gatsby’s  [“Add tags to your `markdown` files”](https://www.gatsbyjs.org/docs/adding-tags-and-categories-to-blog-posts/#add-tags-to-your-markdown-files) uses this format:

```yaml
tags: ["animals", "Chicago", "zoos"]
```

Comment (smockle) [reply docs/content/CounterLabel.md]: Updated in https://github.com/primer/components/pull/586/commits/faee8b0f95d87ea3874399cda87b9363447fd4cb.

File: docs/content/CounterLabel.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ---
 title: CounterLabel
+tags: ["pill"]
```
Comment (emplums) [inline docs/content/CounterLabel.md]: Sorry, these don't need to be in an array, you can list it in this format:
```suggestion
tags: pill
```

File: docs/content/StyledOcticon.md [hunk 3]
```diff
@@ -1,8 +1,9 @@
 ---
 title: StyledOcticon
+tags: ["icon"]
```
Comment (emplums) [inline docs/content/StyledOcticon.md]:
```suggestion
tags: icon
```

File: docs/content/CounterLabel.md (hunk 2 above)
Comment (smockle) [reply docs/content/CounterLabel.md]: Awesome, thanks! I was just trying to follow the [Gatsby docs](https://github.com/primer/components/pull/586#discussion_r334171629), but if this works too, even better. :)

**Pull request merged**